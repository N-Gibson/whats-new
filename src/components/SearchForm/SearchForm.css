* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

nav {
  display: flex;
  flex-direction: column;
  grid-area: nav;
  height: 50px;
}

.search {
  display: inline;
}