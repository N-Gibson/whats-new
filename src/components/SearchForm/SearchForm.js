import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
  }

  search(event) {
    const inputValue  = event.target.parentNode.firstChild.value.toLowerCase();
    this.props.searchNews(inputValue)
  }

  render() {
    return (
      <div className='search'>
        <input className='search-input' type='text' onKeyDown={event => this.props.onEnter(event)}></input>
        <button className='search-button' onClick={event => this.search(event)}>Search Now</button>
      </div>
    )
  }
}

export default SearchForm;