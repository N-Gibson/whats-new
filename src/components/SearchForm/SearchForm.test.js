import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('Search Form', () => {
  let searchNews, onEnter, wrapper
  beforeEach(() => {
    searchNews = jest.fn();
    onEnter = jest.fn();
    wrapper = shallow(<SearchForm searchNews={searchNews} onEnter={onEnter}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to search', () => {
    wrapper.instance().search = jest.fn();
    const mockEvent = { target: { parentNode: { firstChild: { value: 'finally the value '} } } };

    wrapper.find('button').simulate('click', mockEvent);
    
    expect(wrapper.instance().search).toHaveBeenCalledWith(mockEvent);
  });

  
  it('should be able to run functions on enter', () => {
    const mockEvent = { target: { parentNode: { firstChild: { value: 'finally the value '} } } };
    const mockRunOnEnter = jest.fn();

    wrapper.find('input').simulate('onKeyDown', mockEvent);

    expect(mockRunOnEnter).toHaveBeenCalledWith(mockEvent)
  });
});