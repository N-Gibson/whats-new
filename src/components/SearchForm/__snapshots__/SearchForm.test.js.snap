// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form should match snapshot 1`] = `
<div
  className="search"
>
  <input
    className="search-input"
    onKeyDown={[Function]}
    type="text"
  />
  <button
    className="search-button"
    onClick={[Function]}
  >
    Search Now
  </button>
</div>
`;
